File: day2/2.cc
#include<iostream>
#include<string>
#include<fstream>
#include<vector>
#include<sstream>
#include<algorithm>

int main () {

    std::ifstream inp("input.txt");
    std::string line;
    
    int out = 0;

    while(std::getline(inp,line)) {

        std::stringstream ss(line);
        std::string tmp;
        std::vector<std::string> words;

        while(std::getline(ss,tmp,' ')) {
            words.push_back(tmp);
        }

        size_t split_point = words[0].find("-");

        int pos1 = std::stoi(words[0].substr(0,split_point));
        int pos2 = std::stoi(words[0].substr(split_point+1));
    
        std::string password;
        char ch;
        ch = words[1][0];
        password = words[2];

        if((ch == password.at(pos1-1)) ^ (ch == password.at(pos2-1))) out += 1;

    }

    std::cout << out << std::endl;
    return 0;

}

File: day2/2.py
with open("input.txt") as f:
    inp = f.readlines()

inp = list(map(lambda x: x.split(), inp))

def solve(line):
    
    pos1, pos2 = line[0].split('-')
    pos1, pos2 = int(pos1), int(pos2)

    char = line[1][0]

    return (line[2][pos1-1] == char) ^ (line[2][pos2-1] == char)

if __name__=="__main__":
    print(sum(list(map(solve, inp))))